crafting-dead-immerse/src/main/java/com/craftingdead/immerse/game/survival/SurvivalPlayerHandler.java

```diff
@@ -93,6 +99,17 @@ public boolean handleKill(Entity target) {
     return false;
   }
 
+  @Override
+  public boolean handleBlockBreak(BlockPos pos, BlockState block, MutableInt xp) {
+    if (block.is(ImmerseBlocks.BASE_CENTER.get())
+        && LevelExtension.getOrThrow(this.player.getLevel()).getLandManager().getLandOwnerAt(pos).orElse(null) instanceof LegacyBase base
+        && !base.getOwnerId().equals(this.player.getEntity().getUUID())) {

```

Comment:
Check that they're within the build range of the base too

Reply:
Yeah because there’s only a certain area you can modify blocks with the base claim. The inner area is buildable but the outer area is not (acts as a buffer zone between bases).

Reply:
The current code also has an unintended side effect of all base blocks without a land be only breakable by the owner of the valid base, I changed the way I grab the base owner to check but it looks a little over complicated